src/test/java/racingcar/model/CarTest.java

```diff
@@ -0,0 +1,104 @@
+package racingcar.model;
+
+import static org.assertj.core.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+import racingcar.model.RacingGameTest.FixedMoveStrategy;
+import racingcar.model.car.Car;
+import racingcar.model.car.MoveStrategy;
+import racingcar.model.car.RandomMoveStrategy;
+
+class CarTest {
+    public static class FixedStopStrategy implements MoveStrategy {
+        @Override
+        public boolean canMove() {
+            return false;
+        }
+    }
+
+    private Car createCar(String carName) {
+        return new Car(carName, new RandomMoveStrategy());
+    }
+
+    @Test
+    void 이름사이에_공백포함시_예외() {
+        // given
+        String carName = "정 년이";
+
+        // when
+        // then
+        assertThatThrownBy(() -> createCar(carName))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("자동차 이름에 공백이 포함될 수 없습니다.");
+    }
+
+    @Test
+    void 한글_영어만_자동차이름으로_성공() {
+        // given
+        String carName = "정년이";
+
+        // when
+        // then
+        assertThat(createCar(carName).getCarName()).isEqualTo(carName);
+    }
+
+    @Test
+    void 한글_영어만_자동차이름으로_실패() {
+        // given
+        String carName = "123";
+
+        // when
+        // then
+        assertThatThrownBy(() -> createCar(carName))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("자동차 이름은 한글과 영어만 포함해야합니다.");
+    }
+
+    @Test
+    void 자동차이름은_5글자_이하일때_성공() {
+        // given
+        String carName = "햇빛";
+
+        // when
+        // then
+        assertThat(createCar(carName).getCarName()).hasSameSizeAs(carName);
+    }
+
+    @Test
+    void 자동차이름은_5글자_초과일때_실패() {
+        // given
+        String carName = "미스터션샤인";
+
+        // when
+        // then
+        assertThatThrownBy(() -> createCar(carName))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("자동차 이름은 5자 이하여야 합니다.");
+    }
+
+    @Test
+    void 자동차가_이동() {
+        // given
+        String carName = "볼보";
+
+        // when
+        Car car = new Car(carName, new FixedMoveStrategy());
```

Comment:
전략 패턴을 활용하여 테스트해보기 위해선 꼭 새로운 클래스(FixedMoveStrategy)를 만들어야 할까요?
함수형 인터페이스에 대해 학습해 보시면 좋을 것 같아요~

Reply:
와 감사합니다.  이건 확실히 함수형 인터페이스를 사용하는게 좋을 것 같습니다. 👍
다음에 전략패턴을 사용한다면 꼭 함수형 인터페이스를 사용하겠습니다.
```suggestion
        Car car = new Car(carName, () -> true);
```
or
```suggestion
        MoveStrategy FixedMoveStrategy = () -> true;
        Car car = new Car(carName, FixedMoveStrategy);
```